tests/extract/test_validation.py

```diff
@@ -0,0 +1,126 @@
+"""Tests for resource validation with pydantic schema
+"""
+import typing as t
+
+import pytest
+import dlt
+from dlt.common.schema.typing import ColumnValidator
+from dlt.extract.validation import PydanticValidator
+from dlt.extract.exceptions import ValidationError, ResourceExtractionError
+
+from pydantic import BaseModel
+
+
+class SimpleModel(BaseModel):
+    a: int
+    b: str
+
+
+def test_validator_model_in_decorator() -> None:
+    # model passed in decorator
+    @dlt.resource(columns=SimpleModel)
+    def some_data() -> t.Iterator[t.Dict[str, t.Any]]:
+        yield {"a": 1, "b": "2"}
+        yield {"a": 2, "b": "3"}
```

Comment:
also yield a list of objects

Reply:
All tests parametrized now to test both single item and list